src/transport_ofi.h

```diff
@@ -299,18 +299,20 @@ void shmem_transport_put_quiet(void)
         success = fi_cntr_read(shmem_transport_ofi_put_cntrfd);
         fail = fi_cntr_readerr(shmem_transport_ofi_put_cntrfd);
 
-        if (success < shmem_transport_ofi_pending_put_counter && fail == 0) {
+        if (success < \
+            shmem_internal_atomic_read(&shmem_transport_ofi_pending_put_counter) && \
+            fail == 0) {
             SPINLOCK_BODY();
         }
         else if (fail) {
             struct fi_cq_err_entry e = {0};
             fi_cq_readerr(shmem_transport_ofi_put_nb_cqfd, (void *)&e, 0);
             OFI_CQ_ERROR(shmem_transport_ofi_put_nb_cqfd, &e);
         }
-    } while (success < shmem_transport_ofi_pending_put_counter);
+    } while (success < shmem_internal_atomic_read(&shmem_transport_ofi_pending_put_counter));
 #else
     int ret = fi_cntr_wait(shmem_transport_ofi_put_cntrfd,
-                           shmem_transport_ofi_pending_put_counter, -1);
+                           shmem_internal_atomic_read(&shmem_transport_ofi_pending_put_counter), -1);
```

Comment:
I think we need to check if the pending put counter increased while we were blocked.  I'm thinking of something like:
```
uint64_t cnt, cnt_new;
cnt_new = shmem_internal_atomic_read(&shmem_transport_ofi_pending_put_counter), -1);
do {
  int ret;
  cnt = cnt_new;
  ret = fi_cntr_wait(shmem_transport_ofi_put_cntrfd, &cnt, -1);
  ...
  cnt_new = shmem_internal_atomic_read(&shmem_transport_ofi_pending_put_counter), -1);
} while (cnt != cnt_new);
```